frontend/src/components/contracts/ContractDetails.tsx

```diff
@@ -0,0 +1,147 @@
+import moment from "moment";
+
+import React from "react";
+
+import { Row, Col } from "react-bootstrap";
+
+import ContractsApi from "../../libraries/explorer-wamp/contracts";
+
+import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+
+interface Props {
+  accountId: string;
+}
+
+interface State {
+  locked?: boolean;
+  transactionHash?: string;
+  timestamp?: number;
+  codeHash?: string;
+}
+
+export default class extends React.Component<Props, State> {
+  state: State = {};
+
+  collectContractInfo = async () => {
+    new ContractsApi()
+      .getContractInfo(this.props.accountId)
+      .then(contractInfo => {
+        if (contractInfo) {
+          this.setState({
+            codeHash: contractInfo.codeHash,
+            transactionHash: contractInfo.transactionHash,
+            timestamp: contractInfo.timestamp,
+            locked: contractInfo.accessKeys.every(
+              (key: any) =>
+                key["access_key"]["permission"]["FunctionCall"] !== undefined
+            )
+          });
+        }
+        return;
+      })
+      .catch(err => console.error(err));
+  };
+
+  componentDidMount() {
+    this.collectContractInfo();
+  }
+
+  componentDidUpdate(preProps: Props) {
+    if (this.props.accountId !== preProps.accountId) {
+      this.collectContractInfo();
+    }
+  }
+
+  render() {
+    const { locked, transactionHash, timestamp, codeHash } = this.state;
+    let lockedShow;
+    if (locked !== undefined) {
+      lockedShow = locked === true ? "Yes" : "No";
+    }
+    console.log(codeHash, lockedShow);
+    return transactionHash ? (
```

Comment:
It is better to just return earlier as it is really hard to parse these long conditions.

Reply:
what do you mean? sorry I do not get it